fplus-lib/src/core/allocator/mod.rs

```diff
@@ -664,18 +688,18 @@ pub async fn create_allocator_from_file(files_changed: Vec<String>) -> Result<()
                     })?;
 
                 if let Some(allocation_amount) = model.application.allocation_amount.clone() {
-                    let allocation_amounts = allocation_amount.quantity_options.unwrap();
-
-                    for allocation_amount in allocation_amounts {
-                        let parsed_allocation_amount = allocation_amount.replace('%', "");
-                        create_allocation_amount(allocator_id, parsed_allocation_amount)
-                            .await
-                            .map_err(|e| {
-                                LDNError::New(format!(
-                                    "Create allocation amount rows in the database failed: {}",
-                                    e
-                                ))
-                            })?;
+                    if let Some(allocation_amounts) = allocation_amount.quantity_options {
```

Comment:
What if `allocation_amount.quantity_options` is `None`? Is it acceptable to ignore or should we return an error? Previously it would panic, which suggests we should return an error.